### PR TITLE
xtask: Use metadata.build.target when checking individual packages.

### PR DESCRIPTION
**Pull request by flihp**
The docs for xtask state that if --target is not provided then the value
from package.metadata.build.target will be used. This is true when
checking --all packages but not individual packages. This patch
implements this behavior when xtask check is:
- executed from the workspace root with the '-p' option
- executed from a project directory with no params
In both of these cases cargo metadata json is now parsed to extract the
project.metadata.build.target value. If found the target is then added
to the final check command invocation. Otherwise the 'cargo check'
default target (the host architecture) is used.

### CODE REVIEW

File: build/xtask/src/check.rs [hunk 1]
```diff
@@ -3,6 +3,34 @@ use std::process::Command;
...
+
+    #[derive(Debug, Deserialize)]
+    struct BuildMetadata {
+        target: Option<String>,
+    }
+
+    let m: Option<CustomMetadata> =
+        serde_json::from_value(package.metadata).unwrap();
+
+    (|| m?.build?.target)()
```
[the remark this answers is not in the document]
Comment (cbiffle) [reply build/xtask/src/check.rs:31]: The types here suggest that this is equivalent to `m?.build?.target` alone.

**Comment by cbiffle** [comment]
@flihp I'd like to get this merged -- when you get some free time, have a look at @steveklabnik's review comments above, both should be pretty compact fixes.

**Comment by flihp** [comment]
Apologies for the slow action on this. Got buried in my notifications. Should have all review feedback included.

**Comment by flihp** [comment]
last push had a failed check, just rebased on master to hopefully get better results

**Pull request merged**